### ISSUE TITLE
Make the error state of the Edit display name input more descriptive

### ISSUE THREAD

**Issue by timng-bcgov** [bcgov/api-services-portal#1176]
# API Services Portal Issue

## User Story

**As an** API provider,<br>
**I want** to understand the requirements of a Gateway display name when editing it in the Services Portal<br>
**so that** I can successfully update the Gateway display name.

### ENV

- [x] `DEV`
- [x] `TEST`
- [ ] `PROD`

### TESTCASE

Create a Gateway, navigate to the details page (**My Gateways** > **Gateway**) and select the Edit button beside the current display name. Interact with the form input and provide an updated display name.

### EXPECTED

Inputting a display name that does not satisfy the requirements results in an error state and an input description with the requirements to appear.

### ACTUAL

Providing a display name (>3 characters) that does not satisfy the requirements results in the following toast message only after the form is submitted:
![image](https://github.com/user-attachments/assets/0758fa15-bbc8-4dcf-89db-c3bb0f46028a)

No updates to the form input are present.

## Notes

The error state is currently displayed when the input name is less than 3 characters.

![image](https://github.com/user-attachments/assets/c250522f-120f-4bf8-bfe8-aa9cee804c81)

The **Save** action in the dialog is also disabled.

![image](https://github.com/user-attachments/assets/e8d32b42-a322-4831-8dd5-a1c5f29fbc49)

## Save button is invisible on hover when disabled
Another issue. See image above

## Cancel does not clear
Another small issue to address: if you click cancel, it does not clear the text in the input. Expected would be to reset the input to the current display name.

**Comment by rustyjux** [comment]
so is the missing piece more validation (like to include special characters) in the front end?

**Comment by timng-bcgov** [comment]
I think so, looks like there is only validation for >3 characters in the component right now.

**Comment by timng-bcgov** [comment]
And my screenshot also conveniently illustrates something weird with the hover state of the disabled **Save** button, though I'm not sure what's going on as it looks like it just uses the isDisabled prop from the Chakra button.

**Comment by rustyjux** [comment]
> And my screenshot also conveniently illustrates something weird with the hover state of the disabled **Save** button (it's invisible?), though I'm not sure what's going on as it looks like it just uses the isDisabled prop from the Chakra button.

Whoa I didn't even understand what you were showing until I looked at it! Oddly this only happened once for me, and now it's the expected:
![image](https://github.com/user-attachments/assets/4647803f-8395-4102-a50d-83f2f36e9cba)

I bet that is a NextJS (server side) error. I can't recreate it even in a private window.

Another bug with the edit dialog (which I will add to the description) - if you click cancel, it does not clear the text in the input.

**Comment by rustyjux** [comment]
@Elson9 @ikethecoder oops, I'm bad at Git and made some commits on the wrong branch. I was then trying to cherry pick them over to a new feature branch but they ended up on `dev`. 🤦‍♂️

@timng-bcgov please also review on dev. Should be live shortly.

**Comment by Elson9** [comment]
Tseted out in dev, looks good
